src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -974,7 +980,7 @@ FileMapInfo* MetaspaceShared::open_static_archive() {
   return mapinfo;
 }
 
-FileMapInfo* MetaspaceShared::open_dynamic_archive() {
+FileMapInfo* MetaspaceShared::open_dynamic_archive(bool &has_failed) {
```

Comment:
Shouldn't the null return suffice for failure checking, possibly in combination with a flag check?